File: String/isPlaindrome/isPalindrome.cpp
#include <iostream>
#include <string>

bool is_palindrome(const std::string &str) {
	unsigned b = 0;
	unsigned e = str.size() - 1;
	while (b < e) {
		if (str[b] != str[e]) return false;
		++b; --e;
	}
	
	return true;
}

int main() {
	std::cout << "Is \"racecar\" a palindrome: " 
				<< (is_palindrome("racecar") ? "true":"false") 
				<< std::endl; 
				
	std::cout << "Is \"abccba\" a palindrome: " 
				<< (is_palindrome("abccba") ? "true":"false") 
				<< std::endl; 
				
	std::cout << "Is \"nope\" a palindrome: " 
				<< (is_palindrome("nope") ? "true":"false") 
				<< std::endl; 
}

File: String/isPalindrome/isPalindrome.go
package main

import (
	"fmt"
)

func isPalindrome(str string) bool {
	front := 0
	back := len(str) - 1
	for front < back {
		if str[front] != str[back] {
			return false
		}
		front++
		back--
	}
	return true
}

func main() {
	isPalindromePrint("racecar") // true
	isPalindromePrint("abccba")  // true
	isPalindromePrint("nope")    // false
}

func isPalindromePrint(str string) {
	fmt.Printf("Is \"%s\" a palindrome: %t\n", str, isPalindrome(str))
}
